File: UVa725/.py
cnt = 0
sol = set()
while True:
    n = int(input())
    if n == 0:
        break
    if cnt > 0:
        print()
    cnt += 1
    有解 = False
    for i in range(1234, 10_0000//n+1):
        sol.clear()
        sol.update(str(i).zfill(5), str(i*n))
        if len(sol) == 10:
            print('%d / %05d = %d' % (i*n, i, n))
            有解 = True
    if not 有解:
        print('There are no solutions for %d.' % (n))

File: UVa725/.cpp
#pragma GCC optimize(3, "Ofast", "inline")
#include <bits/stdc++.h>
#define all(x) (x).begin(), (x).end()
using namespace std;
int cnt = 0;
set<char> sol;
int main()
{
    int n;
    while (cin >> n && n)
    {
        if (cnt)
        {
            cout << endl;
        }
        cnt++;
        bool has_sol = false;
        for (size_t i = 1234; i < 100000 / n + 1; i++)
        {
            sol.clear();
            string tmp = to_string(i);
            if (tmp.size() < 5)
            {
                tmp += "0";
            }
            tmp += to_string(i * n);
            sol.insert(all(tmp));
            if (sol.size() == 10)
            {
                printf("%d / %05d = %d\n", i * n, i, n);
                has_sol = true;
            }
        }
        if (!has_sol)
        {
            printf("There are no solutions for %d.\n", n);
        }
    }
    return 0;
}
